players.js

```diff
@@ -259,9 +260,43 @@ class Player {
 			this.proxy.PreviousRemote()
 	}
 	activatePlayer(){
-		if(this.desktopApp){
-			let app = Shell.AppSystem.get_default().lookup_app(this.desktopApp);
-			app.activate();
+		let focusedWindow = global.display.get_focus_window();
+		let playerWindow = this._guessAppWindow(this.identity);
+
+		if (!playerWindow)
+			return
+
+		if (focusedWindow == playerWindow){
+			if (this.playerWindowMinimized)
+				playerWindow.minimize();
+
+			let apps = Shell.AppSystem.get_default().get_running();
+			apps[1].activate(global.get_current_time());
```

Comment:
This approach *almost* works, but it sometimes fails when using more than two workspaces

Reply:
works fine for me. what is the issue?

Reply:
works fine for me. What is the issue?

Reply:
It works as expected, but it isn't what **I** expected.

* Open an application window on two different workspaces
* Open a player in a third workspace
* Click on the window of workspace one
* Change to workspace two **without** interacting with the window
* Activate the 'open app' action (this works as expected)
* Activate the 'open app' action again (takes you to the first workspace)

Now this works as expected, the focus returns to the last focused window, but in this scenario i would expect 'open app' to bring me back to the workspace i was previously on.

Reply:
hmmm, maybe worth implementing something like this if the previous app's workspace is different from the last workspace: https://github.com/arjan/gnome-shell-go-to-last-workspace/blob/master/extension.js

but we would need to figure out which workspace the previous app is on. we can do it with the windows but I don't know if it's possible with with the app info. Do you know what info we can extract from the apps listed in `Shell.AppSystem.get_default().get_running()`? if it's only info regarding the app, we might not be able to know on which workspace it it, unless we cross check again the windows list somehow... another rabbit hole!

Reply:
ok, this is pretty niche but we could save the workspace and if it's different from the current, we go there (nothing is focussed so we do nothing except switching), otherwise we open the app as before.

Reply:
resolved with latest implementation